src/backend/logger/logger.ts

```diff
@@ -65,221 +81,134 @@ const getTimeStamp = () => {
-  const fmtDate = date.toISOString().replaceAll(':', '_')
-  const newLogFile = join(logDir, `heroic-${fmtDate}.log`)
-  try {
-    openSync(newLogFile, 'w')
-  } catch (error) {
-    logError(
-      `Open ${currentLogFile} failed with ${error}`,
-      LogPrefix.Backend,
-      true
-    )
-  }
-
-  // Clean out logs that are more than a month old
-  try {
-    const logs = readdirSync(logDir)
-    logs.forEach((log) => {
-      if (log.includes('heroic-')) {
-        const dateString = log
-          .replace('heroic-', '')
-          .replace('.log', '')
-          .replaceAll('_', ':')
-        const logDate = new Date(dateString)
-        if (
-          logDate.getFullYear() < date.getFullYear() ||
-          logDate.getMonth() < date.getMonth()
-        ) {
-          unlinkSync(`${logDir}/${log}`)
-        }
-      }
-    })
-  } catch (error) {
-    logError(
-      `Removing old logs in ${logDir} failed with ${error}`,
-      LogPrefix.Backend,
-      true
-    )
+  input: LogInputType,
+  options?: {
+    prefix?: LogPrefix
+    showDialog?: boolean
+    skipLogToFile?: boolean
   }
+) {
+  const text = convertInputToString(input)
+  const messagePrefix = `${getTimeStamp()} WARNING: ${getPrefixString(
+    options?.prefix ?? LogPrefix.Backend
+  )}`
 
-  let logs: createLogFileReturn = {
-    currentLogFile: '',
-    lastLogFile: ''
-  }
-  if (configStore.has('general-logs')) {
-    logs = configStore.get('general-logs') as createLogFileReturn
-  }
-
-  logs.lastLogFile = logs.currentLogFile
-  logs.currentLogFile = newLogFile
-
-  configStore.set('general-logs', logs)
+  console.warn(messagePrefix, ...(Array.isArray(input) ? input : [input]))
 
-  // get longest prefix to log lines in a kind of table
-  for (const prefix in LogPrefix) {
-    if (longestPrefix < String(prefix).length) {
-      longestPrefix = String(prefix).length
-    }
+  if (options?.showDialog) {
+    showErrorBoxModalAuto(options?.prefix ?? LogPrefix.Backend, text)
   }
 
-  return logs
-}
-
-/**
- * Returns according to options the fitting log file
- * @param isDefault   getting heroic default log
- * @param appName     needed to get appName log
- * @param defaultLast if set getting heroic default latest log
- * @returns path to log file
- */
-export function getLogFile(
-  isDefault: boolean,
-  appName: string,
-  defaultLast = false
-): string {
-  return isDefault
-    ? defaultLast
-      ? lastLogFile
-      : currentLogFile
-    : join(heroicGamesConfigPath, appName + '-lastPlay.log')
-}
-
-/**
- * Appends given message to the current log file
- * @param message message to append
- */
-function appendMessageToLogFile(message: string) {
-  try {
-    if (currentLogFile) {
-      appendFileSync(currentLogFile, `${message}\n`)
-    }
-  } catch (error) {
-    logError(`Writing log file failed with ${error}`, LogPrefix.Backend, true)
+  if (!options?.skipLogToFile) {
+    appendMessageToLogFile(`${messagePrefix} ${text}`)
   }
 }
```

Comment:
I feel like I've said this before, but having one common "logBase" function that all of these functions then call with their respective log level and arguments would be easier to read, maintain, and test